src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -1893,5 +1896,35 @@ public Response getMakeDataCountMetric(@PathParam("id") String idSupplied, @Path
         }
     }
 
+    @GET
+    @Path("{id}/externalTools")
+    public Response getExternalTools(@PathParam("id") String idSupplied, @QueryParam("type") String type) {
```

Comment:
I get the idea of moving these to each of the Dataset and File APIs, but there is a lot of redundancy - it could be a nice clean up to have the core of the code centralized.

Reply:
Good idea. In e92c8f12d I reduced the duplicate code, consolidating it.